--- binutils-2.39/bfd/config.bfd	2022-07-08 11:46:47.000000000 +0200
+++ binutils/bfd/config.bfd	2022-08-05 22:24:14.985249680 +0200
@@ -237,6 +237,19 @@
 
 case "${targ}" in
 # START OF targmatch.h
+  i[3-7]86-*-muos*)
+    targ_defvec=i386_elf32_vec
+    targ_selvecs=
+    targ64_selvecs=x86_64_elf64_vec
+    ;;
+    
+#ifdef BFD64
+  x86_64-*-muos*)
+    targ_defvec=x86_64_elf64_vec
+    targ_selvecs=i386_elf32_vec
+    want64=true
+    ;;
+#endif
 #ifdef BFD64
   aarch64-*-darwin*)
     targ_defvec=aarch64_mach_o_vec
--- binutils-2.39/config.sub	2022-07-08 11:46:47.000000000 +0200
+++ binutils/config.sub	2022-08-05 22:24:52.798092457 +0200
@@ -1753,7 +1753,7 @@
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
+	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* | muos* \
-	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
 	     | fiwix* )
 		;;
 	# This one is extra strict with allowed versions
--- binutils-2.39/gas/configure.tgt	2022-07-08 11:46:47.000000000 +0200
+++ binutils/gas/configure.tgt	2022-08-05 22:24:14.985249680 +0200
@@ -263,6 +263,8 @@
   i386-*-*nt*)				fmt=coff em=pe ;;
   i386-*-rdos*)				fmt=elf ;;
   i386-*-darwin*)			fmt=macho ;;
+  
+  i386-*-muos*)				fmt=elf ;;
 
   ia16-*-elf*)				fmt=elf ;;
 
--- binutils-2.39/ld/configure.tgt	2022-07-29 09:37:48.000000000 +0200
+++ binutils/ld/configure.tgt	2022-08-05 22:24:14.985249680 +0200
@@ -80,6 +80,15 @@
 # Please try to keep this table more or less in alphabetic order - it
 # makes it much easier to lookup a specific archictecture.
 case "${targ}" in
+i[3-7]86-*-muos*)
+			targ_emul=elf_i386_muos
+			targ_extra_emuls=elf_i386
+			targ64_extra_emuls="elf_x86_64_muos elf_x86_64"
+			;;
+x86_64-*-muos*)
+			targ_emul=elf_x86_64_muos
+			targ_extra_emuls="elf_i386_muos elf_x86_64 elf_i386"
+			;;
 aarch64_be-*-elf)	targ_emul=aarch64elfb
 			targ_extra_emuls="aarch64elf aarch64elf32 aarch64elf32b armelfb armelf"
 			;;
--- binutils-2.39/ld/emulparams/elf_i386_muos.sh	1970-01-01 01:00:00.000000000 +0100
+++ binutils/ld/emulparams/elf_i386_muos.sh	2022-08-05 22:24:14.985249680 +0200
@@ -0,0 +1,3 @@
+. ${srcdir}/emulparams/elf_i386.sh
+GENERATE_SHLIB_SCRIPT=yes
+GENERATE_PIE_SCRIPT=yes
--- binutils-2.39/ld/emulparams/elf_x86_64_muos.sh	1970-01-01 01:00:00.000000000 +0100
+++ binutils/ld/emulparams/elf_x86_64_muos.sh	2022-08-05 22:24:14.985249680 +0200
@@ -0,0 +1 @@
+. ${srcdir}/emulparams/elf_x86_64.sh
--- binutils-2.39/ld/Makefile.am	2022-07-08 11:46:48.000000000 +0200
+++ binutils/ld/Makefile.am	2022-08-05 22:24:14.985249680 +0200
@@ -281,6 +281,7 @@
 	eelf_i386_ldso.c \
 	eelf_i386_sol2.c \
 	eelf_i386_vxworks.c \
+	eelf_i386_muos.c \
 	eelf_iamcu.c \
 	eelf_s390.c \
 	eh8300elf.c \
@@ -457,6 +458,7 @@
 	eelf_x86_64_fbsd.c \
 	eelf_x86_64_haiku.c \
 	eelf_x86_64_sol2.c \
+	eelf_x86_64_muos.c \
 	ehppa64linux.c \
 	ei386pep.c \
 	emmo.c
